### PR TITLE
Feat: Added Product listing page and refactored code structure

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
Added API call to get products
Created Context for products
Products are displayed from context state
Filters have not been added yet

### CODE REVIEW

File: src/pages/product-listing/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { getProductListService } from "../../services/getProductListService";
+import { useEffect } from "react";
+import { ProductCard } from "../../components/product-card";
+const ProductListing = () => {
+	const { productListingState, productListingDispatch } = useProductListing();
+	useEffect(() => {
+		(async () => {
+			const products = await getProductListService();
+			if (products)
+				productListingDispatch({ type: "LOAD_DATA", payload: products });
```
Comment (pritam-kr) [inline src/pages/product-listing/index.js]: You can store these actions in separate file.

File: src/pages/product-listing/product-listing.css [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+.gap-xs {
+	gap: 0.5rem;
+}
+
+@media (max-width: 1024px) {
+	.grid-30-70 {
+		display: grid;
+		grid-template-columns: 1fr;
+	}
+}
```
Comment (pritam-kr) [inline src/pages/product-listing/product-listing.css:41]: decrease Tab space.

**Review by pritam-kr — APPROVED** [review]
Everything looks good to me. And I have some points.

File: src/services/getProductListService.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
+
+const getProductListService = async () => {
+	const res = await axios.get("/api/products");
+	if (res.status === 200) {
+		return res.data.products;
+	} else {
+		return false;
+	}
+};
```
Comment (himadri2110) [inline src/services/getProductListService.js:10]: Making a separate file for API calls is a nice way of abstraction. Everything looks neat.

**Review by himadri2110 — APPROVED** [review]

**Pull request merged**